core/storage/blog/gae_models.py

```diff
@@ -0,0 +1,335 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the blog data models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import python_utils
+import utils
+(base_models, user_models) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.user])
+
+datastore_services = models.Registry.import_datastore_services()
+
+# Constants used for generating new ids.
+_MAX_RETRIES = 10
+_RAND_RANGE = 127 * 127
+
+
+class BlogPostModel(base_models.BaseModel):
+    """Model to store blog content
+
+    The id of instances of this class has the form
+        [author_id].[generated_string]
+    """
+
+    # The ID of the user the blog is authored by.
+    author_id = datastore_services.StringProperty(required=True)
+    # Title of the blog.
+    title = datastore_services.StringProperty(indexed=True, required=True)
+    # Content of the blog.
+    content = datastore_services.TextProperty(indexed=False, required=True)
+    # The url fragment of the blog.
+    url_fragment = (
+        datastore_services.StringProperty(indexed=True, required=True))
+    # Tags associated with the blog.
+    tags = datastore_services.StringProperty(indexed=True, required=True)
+    # The thumbnail filename of the blog.
+    thumbnail_filename = datastore_services.StringProperty(required=True)
+    # Time when the blog post model was published.
+    published_on = (
+        datastore_services.DateTimeProperty(indexed=True))
+    # Time when the blog post model was last updated.
+    last_updated = (
+        datastore_services.DateTimeProperty(indexed=True, required=True))
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize corresponding to a user:
+        author_id field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostRightsModel references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.author_id == user_id
+        ).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as one instance per user."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data corresponding to a user to export: author_id."""
+        return dict(super(cls, cls).get_export_policy(), **{
```

Comment:
Careful: you _must_ provide the actual class name in `super()`, otherwise inheritance will not work properly. Please change to:  
```suggestion
        return dict(super(BlogPostModel, cls).get_export_policy(), **{
```

(we're moving to Python 3 soon where we don't even need this anymore!)

Reply:
Fixed.